workflow/util/util.go

```diff
@@ -331,13 +332,36 @@ func ResumeWorkflow(wfIf v1alpha1.WorkflowInterface, workflowName string) error
 			wf.Spec.Suspend = nil
 			updated = true
 		}
-		// To resume a workflow with a suspended node we simply mark the node as Successful
+
 		for nodeID, node := range wf.Status.Nodes {
 			if node.Type == wfv1.NodeTypeSuspend && node.Phase == wfv1.NodeRunning {
-				node.Phase = wfv1.NodeSucceeded
-				node.FinishedAt = metav1.Time{Time: time.Now().UTC()}
-				wf.Status.Nodes[nodeID] = node
-				updated = true
+				//if we have a selector then we only unsuspend matching nodes
+				stepMatchesSelector := false
+				if stepSelector == nil || stepSelector.Empty() {
+					stepMatchesSelector = true
+				} else {
+					inputsMap := fields.Set{}
+					for _, param := range node.Inputs.Parameters {
+						if param.Value != nil {
+							inputsMap[param.Name] = *param.Value
+						}
+					}
+
+					stepMatchesSelector = stepSelector.Matches(inputsMap)
+				}
+
+				if stepMatchesSelector {
+					//if we don't have a fail message then we set the node to succeeded otherwise we set to failed and update the message
+					if stepFailMessage == "" {
+						node.Phase = wfv1.NodeSucceeded
+					} else {
+						node.Phase = wfv1.NodeFailed
+						node.Message = stepFailMessage
+					}
+					node.FinishedAt = metav1.Time{Time: time.Now().UTC()}
+					wf.Status.Nodes[nodeID] = node
+					updated = true
+				}
```

Comment:
We need an error or warning message outputted to the CLI if the `resume` command doesn't actually resume anything.

Reply:
👍